geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java

```diff
@@ -94,6 +102,34 @@ public RedisSet(int size) {
     return diff;
   }
 
+  private static int setOpStoreResult(RegionProvider regionProvider, RedisKey destinationKey,
+      Set<byte[]> diff) {
+    if (regionProvider.getRedisData(destinationKey).getType() != REDIS_SET) {
+      regionProvider.getDataRegion().remove(destinationKey);
+    }
+
+    boolean destinationKeyExists = regionProvider.getRedisData(destinationKey).exists();
+    if (diff.isEmpty()) {
+      if (destinationKeyExists) {
+        regionProvider.getDataRegion().remove(destinationKey);
+      }
+      return 0;
+    }
+
+    if (destinationKeyExists) {
+      RedisSet destinationSet = regionProvider.getTypedRedisData(REDIS_SET, destinationKey, false);
+
+      destinationSet.members.clear();
+      destinationSet.members.addAll(diff);
+      destinationSet.storeChanges(regionProvider.getDataRegion(), destinationKey,
+          new ReplaceByteArrays(diff));
+    } else {
+      regionProvider.getDataRegion().put(destinationKey, new RedisSet(diff));
```

Comment:
If you know "diff" is a MemberSet then you can add a RedisSet constructor for it that just uses diff instead of creating a new MemberSet.